src/Microsoft.IdentityModel.Logging/IdentityModelEventSource.cs

```diff
@@ -67,6 +67,7 @@ public void WriteCritical(string message)
             }
         }
 
+        [NonEvent]
```

Comment:
Be default, the Write method (since it is returning void and is in a class subclassing EventSource) is assumed to be generating an event. Since in this case, this method is not really generating any event and acting as a helper method, we need the nonevent attribute. Otherwise, EventSource complains and throws an exception. It threw the same exception earlier when I introduced this method but changing the order of the method somehow fixed it. Now, when it came up again, this seems to be the right fix for that.